app/javascript/components/ansible-playbook-edit-catalog-form/helper.js

```diff
@@ -0,0 +1,315 @@
+import { useFieldApi, useFormApi } from '@@ddf';
+import React from 'react';
+import { Button, TextInput, Checkbox, Select, SelectItem } from 'carbon-components-react';
+import { TrashCan32 } from '@carbon/icons-react';
+import { TreeViewRedux } from '../tree-view';
+
+const convertObject = (inputObj) => {
+  const outputArray = [];
+  for (const [key, value] of Object.entries(inputObj)) {
```

Comment:
We have to avoid the for loops as much as possible.
maybe we can make use of the `map` function, like the example below
```
a = [{name: 'aaa', age: 10}, {name: 'bbb', age: 20}, {name: 'ccc', age: 30}]
a.map((item) => ({key: item.name, value: item.age}))
```

Reply:
@jeffibm  inputObj is an object and we are destructuring using `[key, value] of Object.entries(inputObj)`. Map cannot be used directly as the example u have mentioned. Also just wondering on why we cant use the for loop . Any reason for it ??

Reply:
> @jeffibm inputObj is an object and we are destructuring using `[key, value] of Object.entries(inputObj)`. Map cannot be used directly as the example u have mentioned. Also just wondering on why we cant use the for loop . Any reason for it ??

using `maps` will make the code clean and easy to chain them if needed.

Reply:
yeah, got that . even the eslint is throwing errors when we use for loops